### PR TITLE
Scott firebase collection experiments

### PR DESCRIPTION
**Pull request by skillitzimberg**
## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|     | :sparkles: New feature     |
| ✓   | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Description

The most fundamental change is that the path to write items to and read items from the db is `users/user_token/items`.

This path:

- is passed into `FirestoreCollection` in `App.js`. This is where the app is getting its initial list of items;
- used in `AddItemForm.js` to write items to the db.

When a user creates a new list/token a document with the field `user_token` is added to the `users` collection. For ease of use when working in the [Firebase console](https://console.firebase.google.com/project/tcl-6-smart-shopping-list/database/firestore/data~2Fusers) the document id is the user token. This id is used to validate a shared token.

## Related Issue

n/a

## Acceptance Criteria

- If a user has a valid user token in localStorage, they are shown a list items, if any, associated with that token.
- If the user does not have a user token in localStorage, they are directed to either create a new token (Create New List) or enter a token (share a token).
- If the user enters an invalid token they are notified that the token is invalid. They may try again or create a new token.
- Given that a valid user token exists in localStorage a user can add items to a list.
- If the user attempts to add an item that already exists in the list, they are given a notice that the item exists. Nothing is written to the db.

## Testing Steps / QA Criteria

Pull the branch `scott-firebase-collection-experiements`.  
Run `npm install`.
Run `npm start`.

To test token validation these are valid tokens:

- `annex hagen veldt`
- `baggy jay seder`
- `deal focus mixup`
- `snort haze chai`

### CODE REVIEW

**Review by SaraSweetie — APPROVED** [review]
This looks good. I like that an item isn't added to the database if it already matches something stored there.

**Comment by SaraSweetie** [comment]
My only suggestion for improvement would be to reword the Share List part. Maybe `Join a list` and for the error message, it currently says `Invalid token. Please try again or create a new token.` As a user, they might not know what a token is, and instead of create a new token it might be better to use the wording on the button above so the association is to click there. So maybe something like `Invalid entry. Please try again or Create New List`

As a developer, you know what a token is but a new user might be confused.

**Review by espain16 — COMMENTED** [review]
As usual your code seems very straight forward and like it's doing what was described in your PR.

File: src/components/shopping.js [hunk 1]
```diff
@@ -1,40 +1,18 @@
...
-        console.log('Document successfully deleted!');
-      })
-      .catch(function(error) {
-        console.error('Error removing document: ', error);
-      });
-  }
 
+function Shopping(props) {
   return (
-    <FirestoreCollection
```
Comment (stevelikesmusic) [inline src/components/shopping.js]: The joys of deleting code ❤️

**Review by stevelikesmusic — COMMENTED** [review]
I'm digging this @skillitzimberg! Breaking out users into their own collection opens the gate to a greater set of features. There could be some interesting work if we had actual logged in users (like find someone's list by their name—way way out of scope for what we're doing 😂) or name your list. 

@SaraSweetie great comment and suggestion regarding the messaging 👏 That is much friendlier wording.

**Pull request merged**